From 58eb84b276ca3b724d7a9d25bc52f507c35a3f12 Mon Sep 17 00:00:00 2001
From: Igor Russkikh <irusskik@redhat.com>
Date: Fri, 6 Nov 2020 18:37:10 -0500
Subject: [PATCH 053/139] [netdrv] net: atlantic: Signedness bug in
 aq_vec_isr_legacy()

Message-id: <1604687916-15087-54-git-send-email-irusskik@redhat.com>
Patchwork-id: 338479
Patchwork-instance: patchwork
O-Subject: [RHEL8.4 BZ 1857861 053/139] net: atlantic: Signedness bug in aq_vec_isr_legacy()
Bugzilla: 1857861
RH-Acked-by: David Arcari <darcari@redhat.com>
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: http://bugzilla.redhat.com/1857861

commit d41378713eef6a7d9d9c30cb9a6181ad72f69596
Author: Dan Carpenter <dan.carpenter@oracle.com>
Date:   Wed Nov 13 21:31:58 2019 +0300

    net: atlantic: Signedness bug in aq_vec_isr_legacy()

    irqreturn_t type is an enum and in this context it's unsigned, so "err"
    can't be irqreturn_t or it breaks the error handling.  In fact the "err"
    variable is only used to store integers (never irqreturn_t) so it should
    be declared as int.

    I removed the initialization because it's not required.  Using a bogus
    initializer turns off GCC's uninitialized variable warnings.  Secondly,
    there is a GCC warning about unused assignments and we would like to
    enable that feature eventually so we have been trying to remove these
    unnecessary initializers.

    Fixes: 7b0c342f1f67 ("net: atlantic: code style cleanup")
    Signed-off-by: Dan Carpenter <dan.carpenter@oracle.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>

Signed-off-by: Igor Russkikh <irusskik@redhat.com>

Cc: David Arcari <darcari@redhat.com>
Cc: Igor Russkikh <irusskik@redhat.com>
Signed-off-by: Jan Stancek <jstancek@redhat.com>
---
 drivers/net/ethernet/aquantia/atlantic/aq_vec.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/aquantia/atlantic/aq_vec.c b/drivers/net/ethernet/aquantia/atlantic/aq_vec.c
index 6e19e27b6200..f40a427970dc 100644
--- a/drivers/net/ethernet/aquantia/atlantic/aq_vec.c
+++ b/drivers/net/ethernet/aquantia/atlantic/aq_vec.c
@@ -307,8 +307,8 @@ irqreturn_t aq_vec_isr(int irq, void *private)
 irqreturn_t aq_vec_isr_legacy(int irq, void *private)
 {
 	struct aq_vec_s *self = private;
-	irqreturn_t err = 0;
 	u64 irq_mask = 0U;
+	int err;
 
 	if (!self)
 		return IRQ_NONE;
-- 
2.13.6

